assets/app/scripts/controllers/edit/buildConfig.js

```diff
@@ -633,21 +635,31 @@ angular.module('openshiftConsole')
     };
 
     $scope.updateTriggers = function() {
+
+      // Helper for checking webhook objects in the buildConfig spec and creating if missing.
+      function reuseOrCreateWebhook(type) {
+        var webhooks = _.filter($scope.buildConfig.spec.triggers, function(obj) { return obj.type === type})
+        if (_.isEmpty(webhooks)) {
+          webhooks = {
```

Comment:
Well `concat()` will take either array or an object and appends it to the array on which its called.
But I see your point and will update
```
var webhook = {
   type: type
};
webhook[(type === "GitHub") ? "github" : "generic"] = {
    secret: ApplicationGenerator._generateSecret()
};
webhooks.push(webhook);
```
since the `secret` needs to be an object of the lowercase webhook type
```
            {
                "type": "GitHub",
                "github": {
                    "secret": "aaed4af14894bc36"
                }
            }
```

Thanks ! :)